content/docs/2.0/concepts/external-scalers.md

```diff
@@ -0,0 +1,582 @@
++++
+title = "External Scalers"
+weight = 500
++++
+
+While KEDA ships with a set of [built-in scalers](../scalers), users can also extend KEDA through a GRPC service that implements the same interface as the built-in scalers.
+This document describes the external scaler interfaces and how to implement them in Go, Node, and .NET; however for more details on GRPC refer to [the official GRPC documentation](https://grpc.io/docs/)
+
+## Overview
+
+### Built-in scalers interface
+
+Built-in scalers implement one of the following go interfaces:
+
+```go
+type Scaler interface {
+	GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error)
+	GetMetricSpecForScaling() []v2beta2.MetricSpec
+	IsActive(ctx context.Context) (bool, error)
+	Close() error
+}
+
+type PushScaler interface {
+	Scaler
+
+	Run(ctx context.Context, active chan<- bool)
+}
+```
+
+The `Scaler` interface defines 4 methods:
+- `GetMetrics` and `GetMetricsSpecForScaling` that are used to build the HPA definition for that particular scaler.
```

Comment:
What should we expect here? It's not fully clear to me.

Reply:
There is a bit more details about them in the "Implementing" sections 

> `GetMetricSpec` returns the target value for the HPA definition for the scaler. This scaler will define a static target of 10, but the threshold value is often specified in the metadata for other scalers.

https://github.com/ahmelsayed/keda-docs/blob/push-scaler/content/docs/2.0/concepts/external-scalers.md#6-implementing-getmetrics


> `GetMetrics` returns the value of the metric refered to from `GetMetricSpec`, in this example it's `earthquakeThreshold`. Refer to the HPA docs for how HPA calculates `replicaCount` based on metric value and target value.


https://github.com/ahmelsayed/keda-docs/blob/push-scaler/content/docs/2.0/concepts/external-scalers.md#6-implementing-getmetrics


It's easier to explain with an example, but I can expand the description there a bit. how about

> `GetMetricSpec` returns the target value for the HPA definition for the scaler. For more details refer to [Implementing `GetMetricSpec`]()

>  `GetMetrics` returns the value of the metric refered to from `GetMetricSpec`. For more details refer to [Implementing `GetMetrics`]()

?

Reply:
It helps a lot, thanks!

Examples are always welcome but not a must!